app/modules/ws2812.c

```diff
@@ -16,15 +18,27 @@ typedef struct {
   uint8_t values[0];
 } ws2812_buffer;
 
-// Init UART1 to be able to stream WS2812 data
-// We use GPIO2 as output pin
-static void ws2812_init() {
+// Init UART1 to be able to stream WS2812 data to GPIO2 pin
+// If DUAL mode is selected, init UART0 to stream to TXD0 as well
+// You HAVE to redirect LUA's output somewhere else
+static void ws2812_init(lua_State* L) {
+  const int mode = lua_tointeger(L, 1);
```

Comment:
lua_tointeger returns 0 if missing ( which is MODE_SINGLE ;) ) but your solution is far more better, fixing